sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs

```diff
@@ -106,6 +109,27 @@ public override Request CreateRequest()
         // copied from https://github.com/Azure/azure-sdk-for-net/blob/main/common/Perf/Azure.Test.Perf/TestProxyPolicy.cs
         private void RedirectToTestProxy(HttpMessage message)
         {
+            if (_recording.Mode == RecordedTestMode.Record)
+            {
+                switch (_filter())
+                {
+                    case EntryRecordModel.Record:
+                        break;
+                    case EntryRecordModel.RecordWithoutRequestBody:
+                        message.Request.Content = null;
+                        break;
+                    case EntryRecordModel.DoNotRecord:
+                        return;
+                }
+            }
+            else if (_recording.Mode == RecordedTestMode.Playback)
+            {
+                if (_filter() == EntryRecordModel.RecordWithoutRequestBody)
```

Comment:
Can we add a test-proxy feature?

Reply:
Yep, see https://github.com/Azure/azure-sdk-tools/issues/2434

Reply:
I think most of the code here would remain the same, but instead of setting the Content to null, we'd set some request header to indicate that the body should be skipped.